add-on/src/options/forms/gateways-form.js

```diff
@@ -5,6 +5,10 @@ const browser = require('webextension-polyfill')
 const html = require('choo/html')
 const { normalizeGatewayURL } = require('../../lib/options')
 
+// Warn about mixed content issues when changing the gateway
+// https://github.com/ipfs-shipyard/ipfs-companion/issues/648
+const secureContextUrl = /^https:\/\/.*|^http:\/\/127.0.0.1/
```

Comment:
This should include `^http:\/\/\[\:\:1\]/` (IPv6 loopback address) as it’s treated the same way as 127.0.0.1 in browsers.